compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -3775,8 +3869,41 @@ private BLangExpression createConditionForFieldMatchPattern(int index, BLangFiel
         return createVarCheckCondition(matchPattern, tempVarRef);
     }
 
-    private BLangExpression createVarCheckCondition(BLangMatchPattern matchPattern,
-                                                    BLangSimpleVarRef varRef) {
+    private BLangExpression createConditionForFieldBindingPatterns(List<BLangFieldBindingPattern> fieldBindingPatterns,
+                                                                   BLangSimpleVariableDef varDef,
+                                                                   BLangBlockStmt blockStmt,
+                                                                   Location pos) {
+        BLangExpression condition = null;
+        for (int i = 0; i < fieldBindingPatterns.size(); i++) {
+            if (i == 0) {
+                condition = createConditionForFieldBindingPattern(i, fieldBindingPatterns.get(i), varDef, blockStmt);
```

Comment:
There is a small bug here. When fieldBindingPatterns are empty, the condition cannot be null. It is fixed here.

Reply:
I think we can still improve this a bit to avoid having to do the `i == 0` for each iteration.

```java
        BLangExpression condition = ASTBuilderUtil.createLiteral(pos, symTable.booleanType, true);
        
        if (fieldBindingPatterns.isEmpty()) {
            return condition;
        }

        condition = createConditionForFieldBindingPattern(0, fieldBindingPatterns.get(0), varDef, blockStmt);
        
        for (int i = 1; i < fieldBindingPatterns.size(); i++) {
            BLangExpression fieldMatchPatternCondition =
                    createConditionForFieldBindingPattern(i, fieldBindingPatterns.get(i), varDef, blockStmt);
            condition = ASTBuilderUtil.createBinaryExpr(pos, condition, fieldMatchPatternCondition,
                    symTable.booleanType, OperatorKind.AND, (BOperatorSymbol) symResolver
                            .resolveBinaryOperator(OperatorKind.AND, symTable.booleanType, symTable.booleanType));
        }
        return condition;
```

Reply:
Yes, changed